consensus/istanbul/backend/backend.go

```diff
@@ -275,12 +275,12 @@ type Backend struct {
 
 // IsProxy returns if instance has proxy flag
 func (sb *Backend) IsProxy() bool {
-	return sb.config.Proxy
+	return sb.proxiedPeer != nil || sb.config.Proxy
 }
 
 // IsProxiedValidator returns if instance has proxied validator flag
 func (sb *Backend) IsProxiedValidator() bool {
-	return sb.config.Proxied
+	return (sb.proxyNode != nil && sb.proxyNode.peer != nil) || sb.config.Proxied
```

Comment:
Ditto

Reply:
there are scenarios where the proxied peer is not yet connected, or has disconnect from the proxy (during the network rampup, or simple when the validator reboots). During this time the proxy should still identify itself as a proxy i.e. to start the celostats daemon. So either we have a proxied peer connect or we are configured as a proxy makes the node a proxy.

Reply:
Same thing, `(sb.proxyNode != nil && sb.proxyNode.peer != nil)` can only be true if `sb.config.Proxied` is true, so this is the same as just returning `sb.config.Proxied`

Reply:
i rolled back that change